rust/connlib/phoenix-channel/src/lib.rs

```diff
@@ -824,6 +824,8 @@ fn make_request(url: Url, host: String, user_agent: String) -> Request {
     OsRng.fill_bytes(&mut r);
     let key = base64::engine::general_purpose::STANDARD.encode(r);
 
+    let user_agent = user_agent.replace(|c: char| !c.is_ascii(), "");
```

Comment:
Thinking more about this, wdyt of URL-encoding this instead of stripping? Then we can decode in the portal-side, and be able to retain this information.

We don't use the field now, but will in the future for more policy conditions.

Reply:
> Thinking more about this, wdyt of URL-encoding this instead of stripping? Then we can decode in the portal-side, and be able to retain this information.

We can do that but it is a breaking change to what we currently do so we need to add some logic to the portal first to check if the string is URL-encoded and if it is, decode and otherwise use as is.

Reply:
I'd suggest we strip for now to fix the issue and then coordinate around how we can send this over. There are multiple strategies on how we can deal with this breaking change.

Reply:
It's not a breaking change from the portal's perspective - the user agent field will just be upserted with the encoded info.

Decoding all previous user agents will be essentially a no-op since they are not encoded today.

Reply:
https://github.com/firezone/firezone/pull/9727

Reply:
See https://github.com/firezone/firezone/blob/main/elixir/apps/domain/lib/domain/version.ex